### ISSUE TITLE
Command Execution 

### ISSUE THREAD

**Issue by rabeeshcharles** [mscdex/ssh2#1038]
How to Executing a command to create a folder in FTP?.

I have used the below sample code for the folder creation

![image](https://user-images.githubusercontent.com/30102705/126119221-4e2b8ce2-ee75-491a-85fb-d944121bef88.png)

**OutPut**

![image](https://user-images.githubusercontent.com/30102705/126119452-3798cbb0-f135-4820-baf0-528128096f1d.png)

Kindly give me a idea on this...

**Comment by theophilusx** [comment]
You appear to have both a misalignment of expectations regarding the ssh2 module
and some basic conceptual errors. The ssh2 library is a low level library - it
is very powerful, but only provides a low level API. So for example, it does not
provide the ability to filter results or perform things like a recursive
creation of a directory hierarchy. You have to hook all of that functionality up
yourself.

You also need to understand the difference between basic ssh connections i.e.
connecting to a remote host using the ssh protocol, where you obtain a remote
shell where you can execute commands and doing sftp, which is basically
executing the ftp protocol over an ssh connection. These are two different
operations. If you are connecting to a remote sftp server, you are restricted to
the sftp commands supported by the protocol. If you connect to an server using
basic ssh, you get a remote shell where you can issue commands similar to when
you use a normal ssh client.

For the case where you want to create remote directories on an sftp server, you
have to create each of the directories in the path yourself (if they don't
exist). There is no support for supplying a /deep/ directory path and having the
remote server recursively create any necessary directories in the path.
Basically, you need to break the path up into the individual directories and
then loop through each directory from the first one, query to see if it already
exists and if it doesn't create it and then move on to the next one etc.

There are a number of npm packages out there which are based on ssh2, but which
have created a new API which is at a higher level of abstraction than the one
provided by ssh2. Using one of those packages might suit your requirements
better than directly using ssh2 as they might be at the higher level of abstraction you appear to be looking for.

Regards,

Tim

--
*Tim Cross*

/For gor sake stop laughing, this is serious!/

**Comment by mscdex** [comment]
If the server only allows sftp, then you can only use the sftp API that you get with `conn.sftp()`.
